docs/features/logforwarding/outputs/splunk-forwarding.adoc

```diff
@@ -43,5 +49,135 @@ spec:
       outputRefs:
         - splunk-receiver
 ----
+1. `token` - Points to the secret containing the Splunk HEC token used for authenticating requests.
+2. `url` - The base URL of the Splunk instance.
+3. `index` - Optional. The name of the index to send events to. If not specified, the default index defined within Splunk is used. This supports template syntax to allow dynamic per-event values.
+4. `source` - Optional. The source of events sent to this sink. This supports template syntax to allow dynamic per-event values.
+5. `indexedFields` - Optional. Fields to be added to Splunk index.
+6. `compression` - Optional. Compression configuration, available to are: `none`, `gzip`. Default is `none`.
+
+=== `source`
+In Splunk, the `source` field typically identifies the origin of a log event.
+To ensure consistency and meaningful categorization, the `source` value can be dynamically derived from the `log_type` and `log_source` fields,
+following Cluster Log Forwarder's conventions.
+
+=== `indexedFields`
+
+In Splunk, `indexed_fields` are fields that are extracted at *index time*, rather than at *search time*. This means their values are stored directly in the index alongside the raw event data, allowing for *faster search performance* on those fields.
+
+However, because `indexed_fields` increase storage usage, they should be used *sparingly* and only for *high-value fields* that provide significant search benefits.
+
+Since logs processed by Vector are typically *well-structured JSON*, Splunk can dynamically extract fields at search time. As a result, the use of `indexed_fields` is *not necessary by default*. That said, they may be valuable in *specific use cases*, such as *large datasets with frequent queries on specific fields*.
+In such cases, users can configure the required fields manually.
+
+==== Limitations
+
+* Only *top-level fields* are supported.
+* Field names may only include *letters, digits, dashes (`-`)*, and *underscores (`_`)*.
```

Comment:
This is a restriction of the implementation but not the API.  We should state what is allowed in the API.  Additionally,  we should define how we transform the field name.  That does not go here since the header is "limitation".  Maybe we should reconsider the header since the limitations are at the implementation level but this is really defining specifics about "indexed fields"